src/main/java/gmbs/domain/LottoNumbers.java

```diff
@@ -0,0 +1,109 @@
+package gmbs.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+public class LottoNumbers {
+    private static final int LOTTO_COUNT = 6;
+    private static final int MIN_NUMBER = 1;
+    private static final int MAX_NUMBER = 45;
+
+    private static final String DUPLICATE_ERROR = "[ERROR] 로또 개수는 중복이 불가능합니다.";
+    private static final String COUNT_ERROR = "[ERROR] 로또 개수는 " + LOTTO_COUNT + "개로 제한됩니다.";
+
+    private static final List<LottoNumber> candidateLottoNumbers = new ArrayList<>();
+
+    static {
+        for (int i = MIN_NUMBER; i <= MAX_NUMBER; i++) {
+            candidateLottoNumbers.add(LottoNumber.of(i));
+        }
+    }
+
+    private final List<LottoNumber> lottoNumbers;
+
+    public LottoNumbers() {
+        this.lottoNumbers = generateRandomLottoNumbers();
+    }
+
+    public LottoNumbers(List<Integer> numbers) {
+        validateLottoNumbers(numbers);
+        this.lottoNumbers = convertToLottoNumberList(numbers);
+    }
+
+    private List<LottoNumber> generateRandomLottoNumbers() {
+        Collections.shuffle(candidateLottoNumbers);
+        return List.copyOf(candidateLottoNumbers.subList(0, LOTTO_COUNT));
+    }
+
+    private void validateLottoNumbers(List<Integer> numbers) {
+        validateLottoCount(numbers);
+        validateDuplicateCount(numbers);
+    }
+
+    private void validateLottoCount(List<Integer> numbers) {
+        if (numbers.size() != LOTTO_COUNT) {
+            throw new IllegalArgumentException(COUNT_ERROR);
+        }
+    }
+
+    private void validateDuplicateCount(List<Integer> numbers) {
+        int distinctCount = calDistinctCountFromArray(numbers);
+
+        if (numbers.size() != distinctCount) {
+            throw new IllegalArgumentException(DUPLICATE_ERROR);
+        }
+    }
+
+    private int calDistinctCountFromArray(List<Integer> numbers) {
+        return (int) numbers.stream()
+                .distinct()
+                .count();
+    }
+
+    private List<LottoNumber> convertToLottoNumberList(List<Integer> numbers) {
+        return numbers.stream()
+                .map(LottoNumber::of)
+                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
```

Comment:
```suggestion
                .collect(Collectors.toUnmodifiableList());
```
저번 리뷰에 드리지 않았나용? 훨씬 더 명료하게 작성할 수 있습니다.
확인하셨는데, 기존의 방식을 고수하시는 거면 왜 `collectingAndThen(toList(), Collections::unmodifiableList))`을 사용해야하는지 알려주세요~

Reply:
수정했습니다! 저번 리뷰 때 반영한 줄 알았는데 반영이 되어 있지 않았네요!